blockstorage: Drop legacy -txindex check

maflcko opened the pull request:
The only reason for the check was to print a warning about an increase in storage use. Now that 22.x is EOL and everyone should have migrated (or decided to not care about storage use), remove the check.

Also, a move-only commit is included. (Rebased from https://github.com/bitcoin/bitcoin/pull/22242)

Commit:
scripted-diff: Use blocks_path where possible

-BEGIN VERIFY SCRIPT-
  sed -i 's|].chain_path, .blocks.|].blocks_path|g' $(git grep -l chain_path)
-END VERIFY SCRIPT-

Commit:
index: Drop legacy -txindex check

Commit:
move-only: Move CBlockTreeDB to node/blockstorage

The block index (CBlockTreeDB) is required to write and read blocks, so
move it to blockstorage. This allows to drop the txdb.h include from
`node/blockstorage.h`.

Can be reviewed with:

--color-moved=dimmed-zebra  --color-moved-ws=ignore-all-space

Commit:
Fixup style of moved code

Can be reviewed with --word-diff-regex=.

theStack reviewed (commented):
Concept ACK

There are still two instances where blocks_path can be used (sometimes it's quite annoying that python also allows single quote strings):
```
test/functional/feature_pruning.py:        self.prunedir = os.path.join(self.nodes[2].chain_path, 'blocks', '')
test/functional/wallet_backup.py:            shutil.rmtree(os.path.join(self.nodes[2].chain_path, 'blocks'))
```
With my very limited sed skills, I'd just add another line
`sed -i "s|].chain_path, 'blocks'|].blocks_path|g" $(git grep -l chain_path)`
to the scripted-diff (probably also possible with only one).

fanquake commented:
cc @TheCharlatan

Commit:
scripted-diff: Rename CBlockTreeDB -> BlockTreeDB

-BEGIN VERIFY SCRIPT-
 sed -i 's|CBlockTreeDB|BlockTreeDB|g' $( git grep -l CBlockTreeDB )
-END VERIFY SCRIPT-

TheCharlatan commented:
ACK fae405556d56f6f13ce57f69a06b9ec1e825422b, though I lack historical context to really judge the second commit fa8685597e7302fc136f21b6dd3a4b187fa8e251.

I was curious if it might be possible to easily make one of the `CBlockIndex*` in `BlockTreeDB` a `CBlockIndex&`, but the change became a bit sprawling.

stickies-v commented on test/functional/feature_txindex_compatibility.py:
Should we at some point sunset this test completely? Given that the `txindex` construction is completely separate from `BlockTreeDB`, I don't think there's really any compatibility testing happening anymore?

stickies-v replied:
Okay, thx. I quite like adding "# TODO" comments to make it easier to find future cleanup items, but I don't really know how I'd phrase the "when" clause so no concrete suggestion.

stickies-v commented on test/functional/feature_txindex_compatibility.py:
```diff
@@ -55,23 +54,13 @@ def run_test(self):
-            expected_msg="Error: The block index db contains a legacy 'txindex'. To clear the occupied disk space, run a full -reindex, otherwise ignore this error. This error message will not be displayed again.",
-        )
         # Build txindex from scratch and check there is no error this time
         self.start_node(1, extra_args=["-txindex"])
```
Could add these lines to check that if user is upgrading from an older node, they'll still get a helpful message on how to use txindex again (albeit without cleaning up unused space)?
```suggestion
        self.log.info("Check that \"Use -txindex\" error is thrown and -txindex works normally")
        self.start_node(1)
        assert_raises_rpc_error(-5, "Use -txindex", lambda: self.nodes[1].getrawtransaction(txid=spend_utxo["txid"]))
        self.restart_node(1, extra_args=["-txindex"])
```

stickies-v replied:
Yeah you're right, this doesn't add anything. I wanted to check that a user had clear instructions on how to build the new txindex, but they wouldn't have to do anything at all, as they're already expected to keep `-txindex` in startup options every time.

Could explicitly add a check that we're indeed building a new txindex (as opposed to somehow using the legacy one), but that doesn't seem like a likely issue either so may not be worth testing? No strong view, so can be marked as resolved.

```suggestion
        with self.nodes[1].assert_debug_log(expected_msgs=['Syncing txindex with block chain from height 0']):
            self.start_node(1, extra_args=["-txindex"])
```

stickies-v reviewed (approved):
ACK fae405556d56f6f13ce57f69a06b9ec1e825422b

The upgrade path when coming from legacy txindex is straightforward and clear enough, no need to keep this extra check around for such an old version imo.
